library/core/src/main/java/com/google/android/exoplayer2/extractor/ts/DvbSubtitleReader.java

```diff
@@ -30,22 +32,49 @@
  */
 public final class DvbSubtitleReader implements ElementaryStreamReader {
 
-  private final String language;
-  private final List<byte[]> initializationData;
+  private class SubtitleTrack {
+    private final String language;
+    private final List<byte[]> initializationData;
+
+    SubtitleTrack (String language, List<byte[]> initializationData) {
+      this.language = language;
+      this.initializationData = initializationData;
+    }
+  }
 
-  private TrackOutput output;
+  private List<SubtitleTrack> subtitles = new ArrayList<>();
+  private List<TrackOutput> outputTracks = new ArrayList<>();
   private boolean writingSample;
   private int bytesToCheck;
   private int sampleBytesWritten;
   private long sampleTimeUs;
 
   /**
-   * @param language The subtitle language code.
-   * @param initializationData Initialization data to be included in the track {@link Format}.
+   * @param esInfo The Elementary Stream info
    */
-  public DvbSubtitleReader(String language, byte[] initializationData) {
-    this.language = language;
-    this.initializationData = Collections.singletonList(initializationData);
+  public DvbSubtitleReader(TsPayloadReader.EsInfo esInfo) {
+    String language;
+    List<byte[]> initializationData;
+
+    int pos = 2;
+
+    while (pos + 8 <= esInfo.descriptorBytes.length) {
```

Comment:
@ojw28 , OK, I will implement in  `TsExtractor readEsInfo` the parsing of the subtitling descriptor (section 6.2.41 of ETSI 300 468):

```
subtitling_descriptor(){
  descriptor_tag              8 uimsbf
  descriptor_length           8 uimsbf
  for (i= 0;i<N;i++){
    ISO_639_language_code    24 bslbf
    subtitling_type           8 bslbf
    composition_page_id      16 bslbf
    ancillary_page_id        16 bslbf
  }
}
```

what do you think about the language code and language features, we should put whatever each  standard dictates or we should have a common list of language and feature codes and translate to them and even localize them?